src/utilities/tree_sitter_utilities.rs

```diff
@@ -317,6 +333,17 @@ impl TSQuery {
   }
 }
 
+impl Validator for TSQuery {
+  fn validate(&self) -> Result<(), String> {
+    let mut parser = get_ts_query_parser();
+    parser
+      .parse(self.get_query(), None)
+      .filter(|x| number_of_errors(&x.root_node()) == 0)
+      .map(|_| Ok(()))
+      .unwrap_or(Err(format!("Cannot parse - {}", self.get_query())))
```

Comment:
Do we report what exactly we failed to parse? i.e. for the missing parenthesis "Unexpected end of query string, unmatched `(` at position x:y" or similar?

Reply:
At this moment we do not report why the query is incorrect (or any diagnostics around it) 
I will add diagnostics for this as a follow up. I think it just reporting error does not make sense, without diagnostics. Tracking this issue #494